### PR TITLE
Fix irsa fluent bit

### PR DESCRIPTION
**Pull request by allamand**
Issue #, if available:

Description of changes:

When deploying the aws-fluent-bit chart using IRSA (Iam role for Service Account) we generally create the service account appart so that it can include the appropriate role. But we also need the chart to create the proper RBAC needed for the chart to run properly.

I added a new value `rbac.enable` which is true by default and will be test instead of `serviceaccount.enable` when creating RBAC.

it is used with following values.yaml extract:

```yaml
rbac:
  enable: true
serviceAccount:
  create: false
  name: aws-for-fluent-bit
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: stable/aws-for-fluent-bit/templates/configmap.yaml [hunk 1]
```diff
@@ -158,7 +158,7 @@ data:
         {{- if .Values.elasticsearch.tls }}
         TLS             {{ .Values.elasticsearch.tls }}
         {{- end }}
-        {{- if .Values.elasticsearch.retry_limit }}
+        {{- if .Values.elasticsearch.retryLimit }}
```
Comment (jaypipes) [inline stable/aws-for-fluent-bit/templates/configmap.yaml:161]: unrelated change, yes?

File: stable/aws-for-fluent-bit/templates/clusterrole.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.serviceAccount.create -}}
+{{- if .Values.rbac.create -}}
```
Comment (jaypipes) [inline stable/aws-for-fluent-bit/templates/clusterrole.yaml:1]: I would actually just remove the `{{ if }}` guard entirely from the `clusterrole.yaml` and `clusterrolebinding.yaml` templates. None of the other charts have this guard on either their `clusterrole[binding].yaml` or combined `rbac.yaml` template files, FWICT.

File: stable/aws-for-fluent-bit/values.yaml [hunk 3]
```diff
@@ -109,6 +109,8 @@ elasticsearch:
 #     Format template
 #     Template {time} used={Mem.used} free={Mem.free} total={Mem.total}
 
+rbac:
+  create: true
```
Comment (jaypipes) [inline stable/aws-for-fluent-bit/values.yaml:113]: And just remove this new configuration option...

**Review by jaypipes — CHANGES_REQUESTED** [review]
I think I'd rather just remove the if guards entirely...

File: stable/aws-for-fluent-bit/templates/configmap.yaml (hunk 1 above)
Comment (allamand) [reply stable/aws-for-fluent-bit/templates/configmap.yaml:161]: I correct an issue with my last or also

File: stable/aws-for-fluent-bit/templates/clusterrole.yaml (hunk 2 above)
Comment (allamand) [reply stable/aws-for-fluent-bit/templates/clusterrole.yaml:1]: Yes agree. I'll entirely remove them

File: stable/aws-for-fluent-bit/values.yaml (hunk 3 above)
Comment (allamand) [reply stable/aws-for-fluent-bit/values.yaml:113]: Yes

**Review by allamand — COMMENTED** [review]
@jaypipes take into account your suggestion

**Review by jaypipes — APPROVED** [review]
:+1: sorry for the delay on this @allamand!
